Testing part of custom setup broken

After stackage LTS upgrade I catch error:

```
web3-0.6.0.0: configure (lib + test)                                                                                                   
[1 of 2] Compiling Main             ( /home/akru/devel/hs-web3/Setup.hs, /home/akru/devel/hs-web3/.stack-work/dist/x86_64-linux-nix/Cabal-2.0.1.0/setup/Main.o )
                                                                                                                                       
/home/akru/devel/hs-web3/Setup.hs:23:38: error:                                                                                        
    * Couldn't match expected type `Distribution.Types.UnqualComponentName.UnqualComponentName'                                        
                  with actual type `[Char]'                                                                                            
    * In the second argument of `(==)', namely `"live"'
      In the first argument of `(&&)', namely `testName t == "live"'                                                                   
      In the expression: testName t == "live" && testEnabled t                                                                         
   |         
23 |   where isLiveTest t = testName t == "live" && testEnabled t
   |                                      ^^^^^^
             
/home/akru/devel/hs-web3/Setup.hs:23:48: error:                                                                                        
    Variable not in scope: testEnabled :: TestSuite -> Bool                                                                            
   |                                                                                                                                   
23 |   where isLiveTest t = testName t == "live" && testEnabled t                                                                      
   |                                                ^^^^^^^^^^^      
```

cause Cabal API changed: 
https://hackage.haskell.org/package/Cabal-2.0.1.1/docs/Distribution-PackageDescription.html#t:TestSuite
vs
https://downloads.haskell.org/~ghc/7.0.2/docs/html/libraries/Cabal-1.10.1.0/Distribution-PackageDescription.html#t:TestSuite

@iostat, what do you think about this?

--- a/.gitignore
+++ b/.gitignore
@@ -3,3 +3,4 @@
 .detected-contract-addresses
 .DS_Store
 build/
+dist/

--- a/Setup.hs
+++ b/Setup.hs
@@ -1,13 +1,25 @@
-import           Control.Monad                      (void, when)
-import           Data.List                          (isSuffixOf)
-import           Distribution.PackageDescription    (HookedBuildInfo, PackageDescription (testSuites), TestSuite (..))
+{-# LANGUAGE CPP #-}
+
+import           Control.Monad                             (void, when)
+import           Data.List                                 (isSuffixOf)
+import           Distribution.PackageDescription           (HookedBuildInfo, PackageDescription (testSuites),
+                                                            TestSuite (..))
 import           Distribution.Simple
-import           Distribution.Simple.LocalBuildInfo (ComponentName (..), LocalBuildInfo (..))
-import           Distribution.Simple.Setup          (BuildFlags (..), fromFlag)
+import           Distribution.Simple.LocalBuildInfo        (ComponentName (..), LocalBuildInfo (..))
+import           Distribution.Simple.Setup                 (BuildFlags (..), fromFlag)
 import           Distribution.Simple.Utils
-import           Distribution.Verbosity             (Verbosity)
-import           System.Directory                   (makeAbsolute)
-import           System.Environment                 (getEnv, getEnvironment, setEnv)
+import           Distribution.Verbosity                    (Verbosity)
+import           System.Directory                          (makeAbsolute)
+import           System.Environment                        (getEnv, getEnvironment, setEnv)
+
+#if MIN_VERSION_Cabal(2,0,0)
+
+import           Distribution.Types.ComponentName          (ComponentName (CTestName))
+import           Distribution.Types.ComponentRequestedSpec (ComponentRequestedSpec (OneComponentRequestedSpec))
+import           Distribution.Types.PackageDescription     (enabledComponents)
+import           Distribution.Types.UnqualComponentName    (mkUnqualComponentName)
+
+#endif
 
 buildingInCabal :: IO Bool
 buildingInCabal = do
@@ -15,9 +27,20 @@ buildingInCabal = do
   return $ not ("stack" `isSuffixOf` parentProcess)
 
 -- note: this only works in cabal, stack doesn't seem to pass these?
+
+#if defined(MIN_VERSION_Cabal) && MIN_VERSION_Cabal(2,0,0)
+
+willBuildLiveSuite :: PackageDescription -> Bool
+willBuildLiveSuite = not . null . liveTestComponents
+  where liveTestComponents = flip enabledComponents (OneComponentRequestedSpec (CTestName (mkUnqualComponentName "live")))
+
+#else
+
 willBuildLiveSuite :: PackageDescription  -> Bool
 willBuildLiveSuite = any isLiveTest . testSuites
   where isLiveTest t = testName t == "live" && testEnabled t
+
+#endif
 
 main :: IO ()
 main = defaultMainWithHooks simpleUserHooks
@@ -47,6 +70,7 @@ exportStore = ".detected-contract-addresses"
 
 myBuildHook :: PackageDescription -> LocalBuildInfo -> UserHooks -> BuildFlags -> IO ()
 myBuildHook pd lbi uh flags = do
+    print $ buildArgs flags
     inCabal <- buildingInCabal
     let v = fromFlag $ buildVerbosity flags
         args = buildArgs flags

--- a/Setup.hs
+++ b/Setup.hs
@@ -70,7 +70,6 @@ exportStore = ".detected-contract-addresses"
 
 myBuildHook :: PackageDescription -> LocalBuildInfo -> UserHooks -> BuildFlags -> IO ()
 myBuildHook pd lbi uh flags = do
-    print $ buildArgs flags
     inCabal <- buildingInCabal
     let v = fromFlag $ buildVerbosity flags
         args = buildArgs flags

